packages/subgraph/src/Voting.ts

```diff
@@ -39,37 +41,37 @@ export function handleVoteRevealed(event: VoteRevealed): void {
 }
 
 function updateVote(voteId: BigInt, event: ethereum.Event): void {
-  let vote = loadOrCreateVote(voteId, event)
-  let voting = Voting.bind(event.address)
-  let winningOutcome = voting.getWinningOutcome(voteId)
+  const vote = loadOrCreateVote(voteId, event)
+  const voting = Voting.bind(event.address)
+  const winningOutcome = voting.getWinningOutcome(voteId)
   vote.winningOutcome = castOutcome(winningOutcome)
   vote.save()
 }
 
-function loadOrCreateVote(voteId: BigInt, event: ethereum.Event): Vote | null {
+function loadOrCreateVote(voteId: BigInt, event: ethereum.Event): Vote {
   let vote = Vote.load(voteId.toString())
 
   if (vote === null) {
     vote = new Vote(voteId.toString())
     vote.createdAt = event.block.timestamp
   }
 
-  return vote
+  return vote!
 }
 
-function loadOrCreateJurorDraft(draftId: string, disputeRoundId: BigInt, jurorAddress: Address, event: ethereum.Event): JurorDraft | null {
-  let draft = JurorDraft.load(draftId)
+function loadOrCreateGuardianDraft(draftId: string, disputeRoundId: BigInt, guardianAddress: Address, event: ethereum.Event): GuardianDraft {
+  let draft = GuardianDraft.load(draftId)
 
   if (draft === null) {
-    let voting = Voting.bind(event.address)
-    let controllerAddress = voting.getController()
-    let controller = Controller.bind(controllerAddress)
-    let disputeManagerAddress = controller.getDisputeManager()
-    let disputeRoundIdArray = decodeDisputeRoundId(disputeRoundId)
-    draft = createJurorDraft(disputeManagerAddress, disputeRoundIdArray[0], disputeRoundIdArray[1], jurorAddress, event.block.timestamp)
+    const voting = Voting.bind(event.address)
+    const controllerAddress = voting.getController()
+    const controller = Controller.bind(controllerAddress)
+    const disputeManagerAddress = controller.getDisputeManager().value0
+    const disputeRoundIdArray = decodeDisputeRoundId(disputeRoundId)
+    draft = createGuardianDraft(disputeManagerAddress, disputeRoundIdArray[0], disputeRoundIdArray[1], guardianAddress, event.block.timestamp)
   }
 
-  return draft
+  return draft!
```

Comment:
Double checking, in all these cases where we're removing `null` as a potential return type, it's because we've already ensured that the entity was "load or created", and are just cleaning up the types now?

Reply:
Yes, not sure if I missed the `!` functionality before of it's new tbh, but I found it useful :)